lib/service_manager.dart

```diff
@@ -33,21 +34,23 @@ class ServiceConnectionManager {
   ServiceExtensionManager get serviceExtensionManager =>
       _serviceExtensionManager;
 
-  VmService service;
+  VmServiceWrapper service;
   VM vm;
   String sdkVersion;
 
   bool get hasConnection => service != null;
 
   Stream<Null> get onStateChange => _stateController.stream;
 
-  Stream<VmService> get onConnectionAvailable =>
+  Stream<VmServiceWrapper> get onConnectionAvailable =>
       _connectionAvailableController.stream;
 
   Stream<Null> get onConnectionClosed => _connectionClosedController.stream;
 
-  void vmServiceOpened(VmService service, Future<void> onClosed) {
-    service.getVM().then((VM vm) {
+  Future<void> vmServiceOpened(
+      VmServiceWrapper service, Future<void> onClosed) async {
+    try {
+      final VM vm = await service.getVM();
```

Comment:
nit: feel free to just write
1final vm = await service.getVM()1
instead of
`final VM vm = await service.getVM();1

Reply:
done